zsh-window-title.zsh

```diff
@@ -4,139 +4,182 @@
-	ZWT_DEBUG=$__zwt_debug_default
+    ZSH_WINDOW_TITLE_DEBUG=$__zsh_window_title_debug_default
+    ZSH_WINDOW_TITLE_DIRECTORY_DEPTH=$__zsh_window_title_directory_depth_default
+    ZWT_DEBUG=$__zwt_debug_default
 }
 
 __zwt:version() {
-	'builtin' 'emulate' -LR zsh
-	__zwt:debugger
+    'builtin' 'emulate' -LR zsh
+    __zwt:debugger
 
-	'builtin' 'print' zwt $ZWT_VERSION
+    'builtin' 'print' zwt $ZWT_VERSION
 }
 
-
 # zsh-window-title subcommands
 
 __zsh-window-title:debugger() {
-	'builtin' 'emulate' -LR zsh
+    'builtin' 'emulate' -LR zsh
 
-	(( ZSH_WINDOW_TITLE_DEBUG )) && 'builtin' 'print' $funcstack[2]
+    (( ZSH_WINDOW_TITLE_DEBUG )) && 'builtin' 'print' $funcstack[2]
 }
 
 __zsh-window-title:add-hooks() {
-	'builtin' 'emulate' -LR zsh
-	__zsh-window-title:debugger
+    'builtin' 'emulate' -LR zsh
+    __zsh-window-title:debugger
 
-	# update window title before drawing the prompt
-	add-zsh-hook precmd __zsh-window-title:precmd
+    # update window title before drawing the prompt
+    add-zsh-hook precmd __zsh-window-title:precmd
+    add-zsh-hook precmd __zsh-window-title:precmd-update  # New hook for updating title after command
 
-	# update the window title before executing a command
-	add-zsh-hook preexec __zsh-window-title:preexec
+    # update the window title before executing a command
+    add-zsh-hook preexec __zsh-window-title:preexec
+    add-zsh-hook preexec __zsh-window-title:preexec-update  # New hook for updating title before execution
+
+    # update window title when changing directories
+    add-zsh-hook chpwd __zsh-window-title:chpwd-update  # New hook for updating title on directory change
 }
 
 __zsh-window-title:init() {
-	'builtin' 'emulate' -LR zsh
+    'builtin' 'emulate' -LR zsh
 
-	'builtin' 'typeset' -gi ZWT_DEBUG=${ZWT_DEBUG:-$__zwt_debug_default}
+    'builtin' 'typeset' -gi ZWT_DEBUG=${ZWT_DEBUG:-$__zwt_debug_default}
 
-	'builtin' 'typeset' -gi ZSH_WINDOW_TITLE_DEBUG=${ZSH_WINDOW_TITLE_DEBUG:-$__zsh_window_title_debug_default}
+    'builtin' 'typeset' -gi ZSH_WINDOW_TITLE_DEBUG=${ZSH_WINDOW_TITLE_DEBUG:-$__zsh_window_title_debug_default}
 
-	__zsh-window-title:debugger
+    __zsh-window-title:debugger
 
-	'builtin' 'typeset' -gi ZSH_WINDOW_TITLE_DIRECTORY_DEPTH=${ZSH_WINDOW_TITLE_DIRECTORY_DEPTH:-$__zsh_window_title_directory_depth_default}
+    'builtin' 'typeset' -gi ZSH_WINDOW_TITLE_DIRECTORY_DEPTH=${ZSH_WINDOW_TITLE_DIRECTORY_DEPTH:-$__zsh_window_title_directory_depth_default}
 
-	__zsh-window-title:precmd
+    __zsh-window-title:precmd
 
-	'builtin' 'autoload' -U add-zsh-hook
+    'builtin' 'autoload' -U add-zsh-hook
 
-	__zsh-window-title:add-hooks
+    __zsh-window-title:add-hooks
 }
 
 __zsh-window-title:precmd() {
-	'builtin' 'emulate' -LR zsh
-	__zsh-window-title:debugger
-
-	local title=$(print -P "%$ZSH_WINDOW_TITLE_DIRECTORY_DEPTH~")
+    'builtin' 'emulate' -LR zsh
+    __zsh-window-title:debugger
 
-	'builtin' 'echo' -ne "\033]0;$title\007"
```

Comment:
Thanks to your opening this PR I discovered #6, and added the enhancement, and that changed how both this and the "before" line 108 do this. You'll need to rebase off the latest `main`

Reply:
Done!

Reply:
done!